src/Common/NamedCollections/NamedCollectionsFactory.h

```diff
@@ -1,13 +1,17 @@
 #pragma once
+
 #include <Common/NamedCollections/NamedCollections.h>
 #include <Common/NamedCollections/NamedCollectionsMetadataStorage.h>
 #include <Common/logger_useful.h>
+#include <Core/BackgroundSchedulePoolTaskHolder.h>
+#include <boost/noncopyable.hpp>
 
 namespace DB
 {
 class ASTCreateNamedCollectionQuery;
 class ASTDropNamedCollectionQuery;
 class ASTAlterNamedCollectionQuery;
+class BackgroundSchedulePoolTaskHolder;
```

Comment:
This one is odd. You are adding both the include and the forward declaration

Reply:
Yep, it's indeed a wrong forward-declaration. It's not necessary. I'll remove it